File: C#/T1/Project/PetSimulator/Command.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PetSimulator
{
    enum Commands
    {
        Feed = 1,
        Walk,
        Sleep,
        Play,
        Movie,
        Toilet
    }

    class Command
    {
        public static void checkCommands(string command, Player player, Pet pet)
        {
            // Check command from player intput
            int temp;
            while (!int.TryParse(command, out temp) || temp < 1 || temp > 6)
            {
                // Print out error
                Console.ForegroundColor = ConsoleColor.Yellow;
                Console.Write("[GAME] ");
                Console.ResetColor();
                Console.Write("Invalid command!\n");

                Console.ForegroundColor = ConsoleColor.Yellow;
                Console.Write("[GAME] ");
                Console.ResetColor();
                Console.Write("Choose your choice again: ");
                command = Console.ReadLine();
            }

            Commands pb = (Commands)temp;

            switch(pb)
            {
                case Commands.Feed:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Enter Amount (gram): ");
                    string amount = Console.ReadLine();
                    int temp1;
                    while (!int.TryParse(amount, out temp1))
                    {
                        // Print out error
                        Console.ForegroundColor = ConsoleColor.Yellow;
                        Console.Write("[GAME] ");
                        Console.ResetColor();
                        Console.Write("Invalid amount!\n");

                        Console.ForegroundColor = ConsoleColor.Yellow;
                        Console.Write("[GAME] ");
                        Console.ResetColor();
                        Console.Write("Enter your amount again (gram): ");
                        amount = Console.ReadLine();
                    }

                    Console.ForegroundColor = ConsoleColor.Cyan;
                    Console.Write("\n[{0}] ", player.playerName);
                    Console.ResetColor();
                    Console.Write("I choose Feed and {0} grams of food\n\n", temp1);
                    player.feedingPet(pet, temp1);
                    break;

                case Commands.Walk:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Enter Amount (meter): ");
                    amount = Console.ReadLine();

                    while (!int.TryParse(amount, out temp1))
                    {
                        // Print out error
                        Console.ForegroundColor = ConsoleColor.Yellow;
                        Console.Write("[GAME] ");
                        Console.ResetColor();
                        Console.Write("Invalid amount!\n");

                        Console.ForegroundColor = ConsoleColor.Yellow;
                        Console.Write("[GAME] ");
                        Console.ResetColor();
                        Console.Write("Enter your amount again (meter): ");
                        amount = Console.ReadLine();
                    }

                    Console.ForegroundColor = ConsoleColor.Cyan;
                    Console.Write("\n[{0}] ", player.playerName);
                    Console.ResetColor();
                    Console.Write("I choose Walk and {0} meter(s)\n\n", temp1);
                    player.goForWalk(pet, temp1);
                    break;

                case Commands.Sleep:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Enter Amount (hour): ");
                    amount = Console.ReadLine();

                    while (!int.TryParse(amount, out temp1))
                    {
                        // Print out error
                        Console.ForegroundColor = ConsoleColor.Yellow;
                        Console.Write("[GAME] ");
                        Console.ResetColor();
                        Console.Write("Invalid amount!\n");

                        Console.ForegroundColor = ConsoleColor.Yellow;
                        Console.Write("[GAME] ");
                        Console.ResetColor();
                        Console.Write("Enter your amount again (hour): ");
                        amount = Console.ReadLine();
                    }

                    Console.ForegroundColor = ConsoleColor.Cyan;
                    Console.Write("\n[{0}] ", player.playerName);
                    Console.ResetColor();
                    Console.Write("I choose Sleep and {0} hour(s)\n\n", temp1);
                    player.goForSleep(pet, temp1);
                    break;

                case Commands.Play:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Enter Amount (hour): ");
                    amount = Console.ReadLine();

                    while (!int.TryParse(amount, out temp1))
                    {
                        // Print out error
                        Console.ForegroundColor = ConsoleColor.Yellow;
                        Console.Write("[GAME] ");
                        Console.ResetColor();
                        Console.Write("Invalid amount!\n");

                        Console.ForegroundColor = ConsoleColor.Yellow;
                        Console.Write("[GAME] ");
                        Console.ResetColor();
                        Console.Write("Enter your amount again (hour): ");
                        amount = Console.ReadLine();
                    }

                    Console.ForegroundColor = ConsoleColor.Cyan;
                    Console.Write("\n[{0}] ", player.playerName);
                    Console.ResetColor();
                    Console.Write("I choose Play and {0} hour(s)\n\n", temp1);
                    player.goForPlay(pet, temp1);
                    break;

                case Commands.Movie:
                    Console.ForegroundColor = ConsoleColor.Yellow;
                    Console.Write("[GAME] ");
                    Console.ResetColor();
                    Console.Write("Enter Amount (hour): ");
                    amount = Console.ReadLine();

                    while (!int.TryParse(amount, out temp1))
                    {
                        // Print out error
                        Console.ForegroundColor = ConsoleColor.Yellow;
                        Console.Write("[GAME] ");
                        Console.ResetColor();
                        Console.Write("Invalid amount!\n");

                        Console.ForegroundColor = ConsoleColor.Yellow;
                        Console.Write("[GAME] ");
                        Console.ResetColor();
                        Console.Write("Enter your amount again (hour): ");
                        amount = Console.ReadLine();
                    }

                    Console.ForegroundColor = ConsoleColor.Cyan;
                    Console.Write("\n[{0}] ", player.playerName);
                    Console.ResetColor();
                    Console.Write("I choose Movie and {0} hour(s)\n\n", temp1);
                    player.goForMovie(pet, temp1);
                    break;

                case Commands.Toilet:
                    Console.ForegroundColor = ConsoleColor.Cyan;
                    Console.Write("\n[{0}] ", player.playerName);
                    Console.ResetColor();
                    Console.Write("I choose Toilet(s)\n\n");
                    player.goForToilet(pet);
                    break;
            }
        }
    }
}

File: C++/T2/PetSimulator/PetSimulator/Command.cpp
#define NOMINMAX
#include "Command.h"
#include "ConsoleColor.h"

#include <iostream>

using namespace std;

void Command::checkCommand(int command, Player * player, Pet * pet)
{
	ConsoleColor cc;
	// Check command from player intput
	while (cin.fail() || command < 1 || command > 6)
	{
		// Print out error
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Invalid Command\n";

		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Choose your choice again: ";

		// https://stackoverflow.com/questions/5864540/infinite-loop-with-cin-when-typing-string-while-a-number-is-expected
		// clear error state
		cin.clear();
		// discard 'bad' character(s)
		cin.ignore(numeric_limits<streamsize>::max(), '\n');

		cin >> command;
	}

	int amount = 0;
	switch (command)
	{
	case Feed:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Enter Amount (g): ";
		
		cin >> amount;

		// Check command from player intput
		while (cin.fail())
		{
			// Print out error
			cc.Yellow();
			cout << "[Game] ";
			cc.White();
			cout << "Invalid Input\n";

			cc.Yellow();
			cout << "[Game] ";
			cc.White();
			cout << "Enter your amount again: ";

			// clear error state
			cin.clear();
			// discard 'bad' character(s)
			cin.ignore(numeric_limits<streamsize>::max(), '\n');

			cin >> amount;
		}

		cc.Cyan();
		cout << "[" << player->getName() << "]";
		cc.White();
		cout << " I choose Feed and " << amount << " g\n\n";
		player->feedingPet(pet, amount);

		break;

	case Walk:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Enter Amount (m): ";

		cin >> amount;

		// Check command from player intput
		while (cin.fail())
		{
			// Print out error
			cc.Yellow();
			cout << "[Game] ";
			cc.White();
			cout << "Invalid Input\n";

			cc.Yellow();
			cout << "[Game] ";
			cc.White();
			cout << "Enter your amount again: ";

			// clear error state
			cin.clear();
			// discard 'bad' character(s)
			cin.ignore(numeric_limits<streamsize>::max(), '\n');

			cin >> amount;
		}

		cc.Cyan();
		cout << "[" << player->getName() << "]";
		cc.White();
		cout << " I choose Walk and " << amount << " m\n\n";
		player->goForWalk(pet, amount);

		break;

	case Sleep:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Enter Amount (hr): ";

		cin >> amount;

		// Check command from player intput
		while (cin.fail())
		{
			// Print out error
			cc.Yellow();
			cout << "[Game] ";
			cc.White();
			cout << "Invalid Input\n";

			cc.Yellow();
			cout << "[Game] ";
			cc.White();
			cout << "Enter your amount again: ";

			// clear error state
			cin.clear();
			// discard 'bad' character(s)
			cin.ignore(numeric_limits<streamsize>::max(), '\n');

			cin >> amount;
		}

		cc.Cyan();
		cout << "[" << player->getName() << "]";
		cc.White();
		cout << " I choose Sleep and " << amount << " hr\n\n";
		player->goForSleep(pet, amount);

		break;

	case Play:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Enter Amount (hr): ";

		cin >> amount;

		// Check command from player intput
		while (cin.fail())
		{
			// Print out error
			cc.Yellow();
			cout << "[Game] ";
			cc.White();
			cout << "Invalid Input\n";

			cc.Yellow();
			cout << "[Game] ";
			cc.White();
			cout << "Enter your amount again: ";

			// clear error state
			cin.clear();
			// discard 'bad' character(s)
			cin.ignore(numeric_limits<streamsize>::max(), '\n');

			cin >> amount;
		}

		cc.Cyan();
		cout << "[" << player->getName() << "]";
		cc.White();
		cout << " I choose Play and " << amount << " hr\n\n";
		player->goForPlay(pet, amount);

		break;

	case Movie:
		cc.Yellow();
		cout << "[Game] ";
		cc.White();
		cout << "Enter Amount (hr): ";

		cin >> amount;

		// Check command from player intput
		while (cin.fail())
		{
			// Print out error
			cc.Yellow();
			cout << "[Game] ";
			cc.White();
			cout << "Invalid Input\n";

			cc.Yellow();
			cout << "[Game] ";
			cc.White();
			cout << "Enter your amount again: ";

			// clear error state
			cin.clear();
			// discard 'bad' character(s)
			cin.ignore(numeric_limits<streamsize>::max(), '\n');

			cin >> amount;
		}

		cc.Cyan();
		cout << "[" << player->getName() << "]";
		cc.White();
		cout << " I choose Movie and " << amount << " hr\n\n";
		player->goForPlay(pet, amount);

		break;

	case Toilet:
		cc.Cyan();
		cout << "[" << player->getName() << "]";
		cc.White();
		cout << " I choose Toilet\n\n";
		player->goForToilet(pet);
		break;
	}
}
